src/webui/api/isessionmanager.h

```diff
@@ -43,6 +45,6 @@ struct ISessionManager
     virtual ~ISessionManager() = default;
     virtual QString clientId() const = 0;
     virtual ISession *session() = 0;
-    virtual void sessionStart() = 0;
+    virtual void sessionStart(const std::optional<QString> &sessionId = std::nullopt) = 0;
```

Comment:
It should not be in this interface. It is just to keep `AuthController` abstracted from session management implementation.
You need to define `WebApplication::sessionStartImpl(sessionId)` and call it both from `WebApplication::sessionStart()` and for creating API key based session.

Reply:
Updated to the interface you described